io_buffer.c

```diff
@@ -2508,23 +2617,12 @@ rb_io_buffer_read(VALUE self, VALUE io, size_t length, size_t offset)
 static VALUE
```

Comment:
I think documentation needs update (`at least +length+ bytes`, length is optional).

Reply:
I've updated the documentation for `read`/`write`/`pread`/`pwrite` and it should be more consistent: https://github.com/ruby/ruby/pull/7860

Reply:
Cool, it looks alright.